# [Quality control](https://www.youtube.com/watch?v=wpgRjgDMrH4&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=2)

* Softwares: FastQC (Single file for each read) & MultiQC (Combined file for multiple reads)

* Base Qualities: 

    * Phred Quality Score: Q =  -10 log10 (probability that base is wrong)
    * Quality scores are encoded, most popular one is Sanger encoding.

* Per position sequence(A,G,T,C) content (FastQC)

    * k-mers are subsequences of length  k contained within a biological sequence. 

    * The bias in eary positions is due to enrichment of k-mers at 5' end due to use of random hexamers or transposases in the library prep & they can have preference for certain sequences

    * [Stranded Protocol](https://chipster.csc.fi/manual/library-type-summary.html): 
    
        EX. HISAT2 & TSeq (?), is required while aligning or counting the reads.
        
        While you can check if stranded protocol was used by running th FastQ file through the Quality Control Tool - [RNA Seq Strandedness inference and inner distance estimation useing RseQC]

        The Output of RseQC for Single End is 
        
            1. ++, -- :
        
                which means read mapped to + strand, indicates parental gene on + strand and vice versa
            2. +-, -+:

        And for Paired End:
            1. 1++, 1--, 2+-, 2-+ :
                Its same as above but specific for the numbered read
            2. 1+-, 1-+, 2++, 2--



