# [Aligning RNA-seq reads to reference genome](https://www.youtube.com/watch?v=dJdZk-duMZQ&list=PLjiXAZO27elBj3KYi7ACscgOxlNkNOxPc&index=4)


* Our Reads will always have variation wrt to the reference Genome
* Genome will have non-unique sequence & introns while our Reads come from RNA
* Challenge: RNA-Seq needs to be able to map splice junction spanning reads to genome non-contiguosly (partialGene-intron-partialGene -> Complete Gene ??)

* Aligners convert Fasta file to a data structure which is faster to process (BWT index & suffix array ??)

* Splice-aware-aligners:
1. STAR  (fast, high memory) - Human Genome
2. HISTAT2 (fast, high memory while indexing) - Human & Mouse Genome, smaller genomes can be supplied
3. TopHat2 (slow, less memory) - Many Genomes + can supply genome

* Output
1. BAM - Contains Alignments
2. bai - index file (for genome browser)
8. log 


* HISTAT2
    
    * Reference Genome is indexed(BWT FM ??) for fast searching
    * Global Index - Anchor read in genome
    * local Index - good for spliced reads with short anchors
    * It uses splice information collected for earlier reads in the same run 
    * parameters - Genome, Library type(strandedness), hits to report (Note there might be better alignments than reported ones), base quality encoding, intron length range, require longer anchor length (for assembly), soft clipping.

* STAR
    
    * Reference Genome converted to suffix array for fast searching (??)
    * 2-pass mapping process
        1. splice junctions found in 1st pass are inserted into the global index
        2. all reads are remapped in the second pass
        3. this allows more spliced reads to map even though it doesn't increase splice junctions (??)
    * Maximum aligments: more than this none are reported
    * Chipster offers Ensembl GTF file to detect annotated splice junctions (can also supply)

* The Gene transfer format (GTF) is a file format used to hold information about gene structure. It is a tab-delimited text format based on the general feature format (GFF).


* Multiple Files
    1. Align together
    2. For Single end data: Just provide all the files
    3. For paired end data: Generate filenames first and then provide all the files (to ensure they are paired correctly)


* Output (BAM/SAM(Tab delimited version of BAM))

    1. Contains sorting information
    2. Chromosome Reference name & length
    3. program line - program that produced BAM & command
    4. After this each row will corresspond to one alignment (to which chromosome read aligns and position, etc.) [REFER SLIDES]
    5. Read Sequence & Quality Score
    6. Tags (How many positions does it align etc.)


* Mapping Quality - confidence in Reads' aligment
    
    * depends on uniqueness of aligned region
    * length
    * number of mismatch or gaps
    * expressed in Phred scores but not encoded

* CIGAR string

    1. M - mismatch
    2. I - insertion
    3. D - deletion
    4. N - Intron
    5. S - soft clip (ignore these bases)(?)
    6. H - hard clip (ignore and remove these bases)(?)

* Flag (Int) - Compact way of storing various things, (website in slides)

* Tool for Further analysis mentioned in Slides