# [Quantitation using HTSeq](https://www.youtube.com/watch?v=cMRNqCJ-qjs&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=7)

* HTSeq - Count Reads/Genes or Reads/transcripts
* Cufflinks & StringTie - Both of the above
* kallisto & salmon - Reads/transcripts (Dont really need to map Reads to Genome)


* HTSeq: 
    * Input as BAM & GTF file
    * Naming of Both file must be same
    * While Counting READs/Gene it considers a gene as union of all its exons
    * Multimapping reads & ambiguous reads(alignment on different strands) are not counted
    * 3 modes to handle genes which overlap several genes (Union, Intersection strict and non-empty)
    * Stranded or Unstranded Data

* Estimating gene expression at gene level, Isoform switching problem should be kept in mind (Difference in length will give apparent downregulation or upregulation because read count mismatch but in actuallity the gene expression shifted from longer isoform to shorter isoform)(?? Detailed Explanation needed)
