# [Differential expression analysis](https://www.youtube.com/watch?v=5tGCBW3_0IA&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=12)

* We want to figure out which genes differ in the expermental groups

* Statistical Testing:

    * Biological replicates are important
    * Normalization is a must (total reads for each sample may be different)
    * FPKM transformation is not used (??)
    * Multiple Testing Problem

* Softwares:

    * edgeR
    * DESeq
    * DEXSeq - axon level differences
    * Cuffdiff & Ballgown - transcript level differences 
    * etc. (see slides)


* Things from RP:

    * Negative bionomial modelling methods have improve specificity & sensitivity as well as good control of positive errors.

    * Cuffdiff performance has reduced sensitivity & specificity due to some normalization which tries to account for isoform switching

* Differential Gene Expression Analysis
    
    * Normalization:

        1. For library size
        2. RNA composition Effect (Some expression of genes will seem like regulation if other genes increase or decrease)
            
            G1 G2 G3    If we randomly sample we will get more G1 and G2, G3 will seem downregulated

            5 3 1

            10 3 1 

        3. Number of Reads for gene is afected by transcript length & GC content (Bias)

* Normalization by edgeR and DESeq:

    1. Aim to make normalized counts for non-differentially expressed genes similar between samples. (No change in count distribution)
    2. Assumes that most genes are not differentially expressed and the amount of downregulated & upregulated genes are equal
    3. Data is not transformed but use normalization factors on statistical testing.

    DESeq(2): 

    1. Take GM of gene count
    2. Divide gene count by GM
    3. Take median of this -> sample's normalization factor

    edgeR:

    (??)

* Dispersion:

    1. Before Inter-Group comparison for Gene expression, Intra-Group variablity should be known (We need to know how noisy those groups are)

    2. Dispersion = (BCV)^2

    BCV - Genes Biological coefficient of variation

    3. Causes of Variablity in counts:

    Sample-to-Sample Variation(dispersion)
    Uncertainity in measuring expression by couting reads

    4. Estimating dispersion reliably

    Pool information from genes which are expressed on similar level
    Assumes that genes of similar average expression have similar dispersion

    Methods: edgeR and DESeq2

    visualization: Volcano Plots
