# [Differential expression analysis](https://www.youtube.com/watch?v=5tGCBW3_0IA&list=PLjiXAZO27elABzLA0aHKS9chVA2TldoPF&index=12)

* We want to figure out which genes differ in the expermental groups

* Statistical Testing:

    * Biological replicates are important
    * Normalization is a must (total reads for each sample may be different)
    * FPKM transformation is not used (??)
    * Multiple Testing Problem

* Softwares:

    * edgeR
    * DESeq
    * DEXSeq - axon level differences
    * Cuffdiff & Ballgown - transcript level differences 
    * etc. (see slides)


* Things from RP:

    * Negative bionomial modelling methods have improve specificity & sensitivity as well as good control of positive errors.

    * Cuffdiff performance has reduced sensitivity & specificity due to some normalization which tries to account for isoform switching
